api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -205,7 +207,14 @@ public List<BedLayout> getBedLayoutsByLocation(Location location) {
 				bedLayout.setBedUuid(blm.getBed().getUuid());
 				bedLayout.setStatus(blm.getBed().getStatus());
 				bedLayout.setBedType(blm.getBed().getBedType());
-				bedLayout.setBedTagMaps(blm.getBed().getBedTagMap());
+				bedLayout.setBedTagMaps(new HashSet<>());
+				if (blm.getBed().getBedTagMap() != null) {
+					for (BedTagMap bedTagMap : blm.getBed().getBedTagMap()) {
+						if (BooleanUtils.isNotTrue(bedTagMap.getVoided())) {
+							bedLayout.getBedTagMaps().add(bedTagMap);
+						}
+					}
+				}
```

Comment:
By moving this here, we don't need special behavior in the REST converter to exclude voided data, they are excluded at this layer instead.